### PR TITLE
Support clicking file input elements

### PR DESCRIPTION
**Pull request by AndrewSouthpaw**
Currently, input file selectors of the type

```
<input type="file">
```

are not detected by VimFx. I think this may be the simple change needed to detect these elements and make them clickable.

### CODE REVIEW

**Comment by lydell** [comment]
`<input type="file">` are already detected because of `.tabIndex > -1`. You can test it with simple page:

```html
<!doctype html>
<meta charset="utf-8">
<title>input-file</title>
<input type="file">
```

Please open an issue about the specific file input you found on a specific URL that wasn’t detected because of some other thing.

**Comment by lydell** [comment]
Thanks for giving a specific URL where I can reproduce the problem.

Oh, what a sucky problem. The thing is that the green `+ Add files` "button" is a `<span>` with a very large `<input type="file">` inside (much larger than the `<span>`). The `<span>` has `overflow: hidden;` to cut off the overflow. This combination of being too large and cutting off the overflow causes VimFx's heuristics to not find a visible point to click the `<input type="file">`. Then, the element is considered hidden and doesn't get a hint at all.

We _could_ skip that "visible point check" for `<input type="file">`, but then we'd end up with the same problem that Vimium for Chrome has for this particular element: It _does_ get a hint, but it is placed quite a bit outside the green "button", so you don't actually understand that the green "button" _did_ get a hint. So that doesn't really help.

If I had implemented that file upload widget, I'd added `width: 100%; height: 100%;` to the `<input type="file">` (instead of just relying on that the `<input type="file">` _happens_ to be large enough to cover the entire green "button"). That fixes the problem. Not sure why they don't do that, though – perhaps to support some old, bad browser.

In conclusion, I don't really know what to do about this case. I can't think of any good heuristic right now that would find the element, and at the same time not slow the `f` commands down.

**Comment by lydell** [comment]
I found a way.

**Comment by AndrewSouthpaw** [comment]
:) Fantastic. Thank you so much for fixing this issue!
